libsplinter/src/auth/oauth/rest_api/actix/logout.rs

```diff
@@ -0,0 +1,70 @@
+// Copyright 2018-2020 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! The `GET /oauth/logout` endpoint for removing a user's tokens.
+
+use actix_web::HttpResponse;
+use futures::future::IntoFuture;
+
+use crate::auth::oauth::rest_api::AuthUserOperation;
+use crate::protocol;
+use crate::rest_api::{
+    get_authorization_token, ErrorResponse, Method, ProtocolVersionRangeGuard, Resource,
+};
+
+pub fn make_logout_route(remove_access_op: Box<dyn AuthUserOperation>) -> Resource {
+    Resource::build("/oauth/logout")
+        .add_request_guard(ProtocolVersionRangeGuard::new(
+            protocol::OAUTH_LOGIN_MIN,
+            protocol::OAUTH_PROTOCOL_VERSION,
+        ))
+        .add_method(Method::Get, move |req, _| {
+            // Extract access token from request's `Authorization` header
+            Box::new(match get_authorization_token(&req) {
```

Comment:
Don't think we should be getting the header and parsing it here. IMO, we should update the auth middleware to insert the user's identity into the request's extensions and remove the token by OAuth identity. This would require updating the `auth::rest_api::authorize` function to check authorization for this endpoint.

Reply:
Updated the middleware, as suggested. The endpoint now extracts the user's identity from the request extensions and passes that to the `remove_user_tokens` method, rather than the `access_token`